lib/plausible/site/admin.ex

```diff
@@ -39,15 +39,26 @@ defmodule Plausible.SiteAdmin do
       inserted_at: %{name: "Created at", value: &format_date(&1.inserted_at)},
       timezone: nil,
       public: nil,
-      owner: %{value: &get_owner_email/1},
+      owner: %{value: &get_owner/1},
       other_members: %{value: &get_other_members/1},
       limits: %{
         value: fn site ->
-          case site.ingest_rate_limit_threshold do
-            nil -> ""
-            0 -> "🛑 BLOCKED"
-            n -> "⏱ #{n}/#{site.ingest_rate_limit_scale_seconds}s (per server)"
-          end
+          rate_limiting_status =
+            case site.ingest_rate_limit_threshold do
+              nil -> ""
+              0 -> "🛑 BLOCKED"
+              n -> "⏱ #{n}/#{site.ingest_rate_limit_scale_seconds}s (per server)"
+            end
+
+          owner = Enum.find(site.memberships, fn m -> m.role == :owner end)
```

Comment:
Like mentioned elsewhere, with :owner preload this becomes

```suggestion
          owner = site.owner
```